RegressionTests.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE OverloadedStrings #-}
+module RegressionTests where
+
+import Database.HDBC.Sqlite3
+import Database.HDBC
+import System.IO
+import System.Environment
+import System.Exit
+import System.Process
+import Data.Aeson
+import Data.Aeson.Types
+import Data.Maybe
+import qualified Data.ByteString.Lazy.UTF8 as BLU
+import qualified Data.HashMap.Strict as HM
+import Data.Text (Text)
+import qualified Data.Text.IO as TIO
+import Data.List
+
+status_list :: [Text]
+status_list = ["", "syntaxerror", "notequiv", "similar", "correct", "expected", "buggy"]
+
+getScore :: Maybe Text -> Int
+getScore status = fromMaybe 0 $ elemIndex (fromMaybe "" status) status_list
```

Comment:
I'd use only `Text` to represent the status, and make sure at the call site that you handle missing status by converting it to "". Now there are two ways of representing a missing status `Just ""` and `Nothing`. You could do this with a custom ADT too, but I don't think it's worth the effort as it requires code to parse the status from the DB, etc.